testnet/stacks-node/src/config.rs

```diff
@@ -2572,16 +2572,20 @@ impl MinerConfigFile {
             .map(|x| Secp256k1PrivateKey::from_hex(x))
             .transpose()?;
         let pre_nakamoto_mock_signing = mining_key.is_some();
-        let valid_tenure_cost_limit = self
-            .tenure_cost_limit_per_block_percentage
-            .map(|p| p < 100 && p > 0)
-            .unwrap_or(true);
-        if !valid_tenure_cost_limit {
-            return Err(
-                "miner.tenure_cost_limit_per_block_percentage must be between 1 and 100"
-                    .to_string(),
-            );
-        };
+
+        let tenure_cost_limit_per_block_percentage =
+            if let Some(percentage) = self.tenure_cost_limit_per_block_percentage {
+                if (1..=100).contains(&percentage) {
```

Comment:
This `contains` feels much less efficient than just doing `percentage < 100`. It's an unsigned, so we already know it's `> 0`. 

Can we also have a special case for `== 100` and just make that a `None`? That should allow you to delete some checks in other code.

Reply:
Done.